tests/test_qnode.py

```diff
@@ -1398,6 +1348,488 @@ def circuit(a):
+            qml.PhaseShift(c, wires=1)
+            return qml.expval(qml.PauliX(0))
+
+        # construct subcircuits
+        circuit = qml.QNode(circuit, dev)
+        circuit.construct_metric_tensor([1, 1, 1])
+
+        a = 0.432
+        b = 0.12
+        c = -0.432
+
+        # evaluate subcircuits
+        circuit.metric_tensor(a, b, c)
+
+        # first parameter subcircuit
+        assert np.allclose(circuit._metric_tensor_subcircuits[(0,)]['result'], 0.25, atol=tol, rtol=0)
+
+        # second parameter subcircuit
+        res = circuit._metric_tensor_subcircuits[(1,)]['result']
+        expected = np.cos(a)**2/4
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        # third parameter subcircuit
+        res = circuit._metric_tensor_subcircuits[(2,)]['result']
+        expected = (3-2*np.cos(a)**2*np.cos(2*b)-np.cos(2*a))/16
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+    def test_evaluate_diag_metric_tensor(self, tol):
+        """Test that a diagonal metric tensor evaluates correctly"""
+        dev = qml.device('default.qubit', wires=2)
+
+        @qml.qnode(dev)
+        def circuit(a, b, c):
+            qml.RX(a, wires=0)
+            qml.RY(b, wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.PhaseShift(c, wires=1)
+            return qml.expval(qml.PauliX(0))
+
+        a = 0.432
+        b = 0.12
+        c = -0.432
+
+        # evaluate metric tensor
+        g = circuit.metric_tensor(a, b, c)
+
+        # check that the metric tensor is correct
+        expected = np.array([1, np.cos(a)**2, (3-2*np.cos(a)**2*np.cos(2*b)-np.cos(2*a))/4])/4
+        assert np.allclose(g, np.diag(expected), atol=tol, rtol=0)
+
+    def test_evaluate_block_diag_metric_tensor(self, tol):
+        """Test that a block diagonal metric tensor evaluates correctly,
+        by comparing it to a known analytic result as well as numerical
+        computation."""
+        dev = qml.device('default.qubit', wires=3)
+
+        def non_parametrized_layer(a, b, c):
+            qml.RX(a, wires=0)
+            qml.RX(b, wires=1)
+            qml.RX(c, wires=1)
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+            qml.RZ(a, wires=0)
+            qml.Hadamard(wires=1)
+            qml.CNOT(wires=[0, 1])
+            qml.RZ(b, wires=1)
+            qml.Hadamard(wires=0)
+
+        a = 0.5
+        b = 0.1
+        c = 0.5
+
+        @qml.qnode(dev)
+        def final(x, y, z, h, g, f):
+            non_parametrized_layer(a, b, c)
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.RZ(z, wires=2)
+            non_parametrized_layer(a, b, c)
+            qml.RY(f, wires=1)
+            qml.RZ(g, wires=2)
+            qml.RX(h, wires=1)
+            return qml.expval(qml.PauliX(0))
+
+        params = [-0.282203, 0.145554, 0.331624, -0.163907, 0.57662, 0.081272]
+        x, y, z, h, g, f = params
+
+        G = final.metric_tensor(x, y, z, h, g, f)
```

Comment:
Do we not support list or array inputs? Should we?

Reply:
It does 🙂 It takes the exact same form as the user has defined the QNode signature. If the signature takes a list/array, this method also takes a list/array.